### PR TITLE
KEP-3866 kube-proxy nftables to beta

### PR DESCRIPTION
**Pull request by danwinship**
#### What this PR does / why we need it:
Moves kube-proxy nftables mode to beta, and adds infra for using it in GCE-based CI

#### Special notes for your reviewer:

(The KEP says we're going to implement a few more e2e tests before going to beta, but it will be simpler to implement those tests if we go to beta first...)

#### Does this PR introduce a user-facing change?
```release-note
kube-proxy's nftables mode is now beta and available by default. It
requires Linux kernel version 5.13 or later, and can be enabled by
passing `--proxy-mode=nftables` to kube-proxy.

A few features in the nftables mode behave slightly differently than
in the iptables mode:

  - By default, nftables mode only listens for NodePort connections on
    the node's primary interface, and it never listens for NodePort
    connections on 127.0.0.1.

  - If you are running a firewall on the node, nftables kube-proxy
    requires the firewall to be configured properly for service
    proxying; it will not try to poke holes in the firewall itself.

  - The nftables mode does not attempt to work around conntrack bugs
    in the kernel unless you explicitly configure it to.

Consult the documentation
(https://kubernetes.io/docs/reference/networking/virtual-ips/#proxy-mode-nftables)
for more information on how to tell if these differences affect you,
and what configuration options are available to change these behaviors.

Note that while the nftables code should be stable and fully-featured
at this point, it is still in the process of being optimized, and it
is known to have worse performance than iptables mode in a few
situations. (In particular, kube-proxy startup for a given cluster
size will be slower with nftables mode than with iptables mode.)
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3866
```

/kind feature
/sig network
/priority important-soon
/assign @aojea

### CODE REVIEW

**Comment by aojea** [comment]
pull-kubernetes-unit	seems legit
```
{Failed;  === RUN   TestValidateProxyMode
    validation_test.go:865: Expected error: KubeProxyConfiguration.ProxyMode: Invalid value: "non-existing": must be iptables, ipvs or blank (blank means the best-available proxy [currently iptables]), got KubeProxyConfiguration.ProxyMode: Invalid value: "non-existing": must be iptables, ipvs, nftables or blank (blank means the best-available proxy [currently iptables])
--- FAIL: TestValidateProxyMode (0.00s)
```


the e2e-gce has two failres seems flakes, but let's rerun to be sure

**Comment by aojea** [comment]
/lgtm

/assign @BenTheElder @thockin 

for the cluster folder

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1753,24 +1753,35 @@ function prepare-kube-proxy-manifest-variables {
...
 
-    if sudo modprobe -a ip_vs ip_vs_rr ip_vs_wrr ip_vs_sh ${conntrack_module}; then
-      params+=" --proxy-mode=ipvs"
-    else
-      # If IPVS modules are not present, make sure the node does not come up as
-      # healthy.
-      exit 1
-    fi
-  fi
-  params+=" --iptables-sync-period=1m --iptables-min-sync-period=10s --ipvs-sync-period=1m --ipvs-min-sync-period=10s"
```
Comment (BenTheElder) [inline cluster/gce/gci/configure-helper.sh:1773]: huh .... were we depending on this somewhere else like clobbering the --proxy-mode flag later and then depending on these other flags? 😬

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1753,24 +1753,35 @@ function prepare-kube-proxy-manifest-variables {
...
+        # healthy.
+        exit 1
+      fi
+      params+=" --proxy-mode=ipvs --ipvs-sync-period=1m --ipvs-min-sync-period=10s"
+      ;;
+    nftables)
+      # Pass --conntrack-tcp-be-liberal so we can test that this makes the
+      # "proxy implementation should not be vulnerable to the invalid conntrack state bug"
+      # test pass. https://issues.k8s.io/122663#issuecomment-1885024015
+      params+=" --proxy-mode=nftables --conntrack-tcp-be-liberal"
```
Comment (BenTheElder) [inline cluster/gce/gci/configure-helper.sh:1781]: ... should this flag only be set in certain test jobs or something?

ideally most of the time we should be testing with options an end-user would have set, and not test-specific configuration.

**Review by BenTheElder — COMMENTED** [review]
/approve
/hold

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (aojea) [reply cluster/gce/gci/configure-helper.sh:1781]: this has a long history, without that sysctl set,  proxies can hit a bug caused by NAT/conntrack and TCP race , but the solution of installing rules for solving that breaks other more advanced scenarios , so we recommend set the sysctl https://github.com/kubernetes/kubernetes/issues/117924#issuecomment-1708756765

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (aojea) [reply cluster/gce/gci/configure-helper.sh:1773]: `proxy-mode` empty means iptables

**Comment by aojea** [comment]
/hold cancel

Tests were exercised with a temporal commit and there are some PRs waiting on this

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/issues/124379
/retest

**Pull request merged**